services/network/server/infrastructure/output/tcp.client.go

```diff
@@ -0,0 +1,52 @@
+package output
+
+import (
+	"context"
+	"errors"
+	app "github.com/Snowflake-Labs/sansshell/services/network/server/application"
+	"net"
+	"strconv"
+	"strings"
+	"time"
+)
+
+// TCPClient is implementation of [github.com/Snowflake-Labs/sansshell/services/network/server/application.TCPClientPort] interface
+type TCPClient struct {
+}
+
+// CheckConnectivity is used to check tcp connectivity from remote machine to specified server
+func (p *TCPClient) CheckConnectivity(ctx context.Context, hostname string, port uint8, timeoutSeconds uint32) (*app.TCPConnectivityCheckResult, error) {
+	hostToCheck := net.JoinHostPort(hostname, strconv.Itoa(int(port)))
+
+	timeout := time.Duration(timeoutSeconds) * time.Second
+
+	dialer := net.Dialer{Timeout: timeout}
+	conn, err := dialer.DialContext(ctx, "tcp", hostToCheck)
+
+	if err != nil {
+		var failReason string
+		if netErr, ok := err.(net.Error); ok && netErr.Timeout() {
+			failReason = "Connection timed out"
```

Comment:
It would cause proto changes each time we want add new message from backend. In current implementation we could add as much "failReason" on server as we want without releasing new proto schema 

Also not sure how GRPC would handle when we extend proto and deploy new version to the server, but client would use old version, potentially it could cause parsing error. I could check if you still thing it is good idea

Reply:
As long as you do https://protobuf.dev/programming-guides/dos-donts/#unspecified-enum, backwards-compatibility for enums in public interfaces works fine.

Reply:
Done